server/neptune/gateway/pr/signaler.go

```diff
@@ -0,0 +1,115 @@
+package pr
+
+import (
+	"context"
+	"fmt"
+	"github.com/runatlantis/atlantis/server/core/config/valid"
+	"github.com/runatlantis/atlantis/server/events/models"
+	"github.com/runatlantis/atlantis/server/neptune/workflows"
+	"go.temporal.io/sdk/client"
+)
+
+type signaler interface {
+	SignalWithStartWorkflow(ctx context.Context, workflowID string, signalName string, signalArg interface{},
+		options client.StartWorkflowOptions, workflow interface{}, workflowArgs ...interface{}) (client.WorkflowRun, error)
+}
+
+const (
+	Deprecated = "deprecated"
+	Destroy    = "-destroy"
+)
+
+type WorkflowSignaler struct {
+	TemporalClient signaler
+}
+
+type Options struct {
```

Comment:
Options doesn't really make sense here since these are all required. I didn't really like this with the deploy signaler either. Let's just call this Request at this point.